src/utilities/EventEmitter.test.ts

```diff
@@ -0,0 +1,242 @@
+import { it, describe, expect, beforeEach, afterEach, beforeAll } from 'vitest';
+import EventEmitter, { removeListener } from './EventEmitter';
+
+describe('Removes a listener from an array or object of listeners', () => {
+  const listener = () => true;
+  const altListener = () => false;
+  
+  const targetListenerArrayFixture1 = [altListener, listener, altListener];
+  const targetListenerArrayFixture2 = [listener, altListener, listener];
+
+  it('Successfully removes a listener from arrays', () => {
```

Comment:
all the other specs so far in spaces use lowercase instead of title case for `it` calls, I think we should keep that consistent